src/approval/approval.service.ts

```diff
@@ -0,0 +1,34 @@
+import Axios, { AxiosResponse } from 'axios';
+import { IApproverInfo } from "./approvers.interface";
+import { approvalUrl } from '../config';
+import { PhonebookError, UserNotFoundError, ApplicationError } from '../utils/errors';
+
+export class Approval {
+
+    /**
+     * Gets a user approver information from the phonebook 
+     * @param id - the user ID
+     */
+    async getApproverInfo(id: string): Promise<IApproverInfo> {
+        try {
+            const res: AxiosResponse = await Axios.get(`${approvalUrl}/user/${id}/approverInfo`);
+            const info: IApproverInfo = res.data;
+
+            return info;
+        } catch (err) {
+            if (err.response && err.response.status) {
+                const statusCode: number = err.response.status;
+                if (statusCode === 404) {
+                    throw new UserNotFoundError(`The user with id ${id} is not found`);
+                }
+                // Unauthorized
+                if (statusCode === 401) {
+                    throw new ApplicationError(`Request to phonebook wasn't authorized: ${JSON.stringify(err)} `);
+                }
+                throw new PhonebookError(`Error in contacting the user service : ${JSON.stringify(err)}`);
+            } else {
+                throw new ApplicationError(`Unknown Error while contacting the user service : ${err}`);
+            }
```

Comment:
fit the error handling here to the approval-service

Reply:
after check, it's doing proxy to the kartoffel so the errors should be the same. 
when i will get the ip for testing i will be able to know for sure.